docs/developer-guide/views-examples.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Learn about the most used custom view examples of Lite XL.
+---
+
+## Toolbar
+
+### Tutorial
+
+```lua
+-- mod-version:3 -- lite-xl 2.1
+```
+
+This is required to specify Lite XL version compatibility.
+
+The instructions that begin with `require` add necessary libraries.
+
+```lua
+local toolbar = common.merge({
+  -- plugin configuration options go here
+}, config.plugins.build)
+```
+
+The code above contains the (optional) configuration options of the plugin.
+
+The `get_plugin_directory()` function gets the plugin folder path and tells the 
+plugin where to find the icon font.
+The iteration code is used to check if every folder detected by `get_plugin_directory()` 
+actually exists and if the folder doesn't exist, a `nil` value is returned.
+
+`local ToolBar = ToolbarView:extend()` is used to create a new custom View and make it 
+inherit ToolbarView.
```

Comment:
This is so confusing, the explanation with specific function names should go _after_ the code.

Explanation before the codeblock should explain roughly what the code does (often one to two sentences) and leave the detailed explanation after the codeblock, after the reader has read the code.